### PR TITLE
[ErrorRenderer] extend legacy exception for backwards compatibility

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]
We tried hard not doing so but we have to face the truth, that's the only way :)

File: src/Symfony/Component/ErrorRenderer/Exception/FlattenException.php [hunk 1]
```diff
@@ -39,12 +40,12 @@ class FlattenException
     private $file;
     private $line;
 
-    public static function create(\Exception $exception, int $statusCode = null, array $headers = []): self
+    public static function create(\Exception $exception, $statusCode = null, array $headers = []): self
```
Comment (yceruto) [inline src/Symfony/Component/ErrorRenderer/Exception/FlattenException.php]: `: LegacyFlattenException`?

**Review by yceruto — CHANGES_REQUESTED** [review]
:+1: we should also update the `HttpKernel\...\ExceptionListener` class according to this approach.

**Comment by yceruto** [comment]
Just asking: Did you find another BC break? Won't https://github.com/symfony/symfony/pull/33941 be enough to keep the BC?

**Comment by nicolas-grekas** [comment]
> Just asking: Did you find another BC break? Won't #33941 be enough to keep the BC?

see 3.4 and 4.3 in https://travis-ci.org/symfony/symfony/builds

**Comment by xabbuh** [comment]
Status: Needs Review

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**